src/Cms/App.php

```diff
@@ -984,7 +984,7 @@ protected function optionsFromReadyCallback(): array
      * @param string|null $id
      * @param \Kirby\Cms\Page|\Kirby\Cms\Site|null $parent
      * @param bool $drafts
-     * @return \Kirby\Cms\Page|null
+     * @return \Kirby\Cms\Page|\Kirby\Cms\Pages|null
```

Comment:
How can this method return a `Pages` object? I'm sure it can somehow, otherwise Psalm wouldn't have complained, but should it? I think we should make sure that this does not happen, looks like a bug to me.

Reply:
My mistake, I though it's the same as `page()` - but actually here `$id` can only be one page. I think Psalm gets confused cause `->find()` could return also `Pages` but only if multiple parameters/arguments passed. How can we make Psalm understand that since we only pass a single argument, the return will for sure only be of type `Page`?

Reply:
Ah, sure, that will be it! Maybe this can be solved with conditional types: https://psalm.dev/docs/annotating_code/type_syntax/conditional_types/

Reply:
That could work. Have to test if we can make the required `@template` syntax work for getkirby.com auto docs

Reply:
@lukasbestle It seems like Psalm conditions work for types but not number of parameters

Reply:
Maybe it would be possible with multiple template types and `@psalm-param` for "faked" params, but that's probably way too complex.

Alternative: In line 998 before the `return $page` line, there could be `/** @var \Kirby\Cms\Page */`, which would tell Psalm the actual returned value.